challenge/dns01/nameserver.go

```diff
@@ -26,9 +26,17 @@ var defaultNameservers = []string{
 // recursiveNameservers are used to pre-check DNS propagation.
 var recursiveNameservers = getNameservers(defaultResolvConf, defaultNameservers)
 
-// networkStack is used to define which IP stack will be used. The default is
-// both IPv4 and IPv6. Set to "ipv4" for IPv4 only, and "ipv6" for IPv6 only.
-var networkStack = "both"
+type networkStack int
+
+const (
+	defaultNetworkStack networkStack = iota
+	ipv4Only
+	ipv6Only
```

Comment:
Please export the values (but keep the type private):

```suggestion
	DefaultNetworkStack networkStack = iota
	IPv4Only
	IPv6Only
```

This eliminates ambiguities on the call-site (and allows to drastically simplify the setter). Since the type stays private, users can't easily/accidentally define additional valid values (maybe via `reflect.TypeOf`, but that won't be accidental anymore).

Reply:
I'd also have to make the `networkStack` type public. I made these private in order to appease the linter and follow your original suggestion.

Reply:
I'll put this on hold until we figure out the correct approach.